framework/include/utils/FunctionParserUtils.h

```diff
@@ -44,7 +44,7 @@ class FunctionParserUtils
 
 protected:
   /// Evaluate FParser object and check EvalError
-  Real evaluate(ADFunctionPtr &);
+  Real evaluate(ADFunctionPtr &, std::string object_name = "");
```

Comment:
Let's not pass strings by value

Reply:
`const std::string & object_name = ""` should do it right?